src/internal/views.py

```diff
@@ -332,3 +333,74 @@ def has_permission(self):
                 self.request.user.has_perm('internal.delete_quote')
                 or self.request.user == self.get_object().author
         )
+
+
+class LoreListView(ListView):
+    model = Lore
+    ordering = ['title']
+    template_name = 'internal/lore/lore_list.html'
+    context_object_name = 'all_lore_articles'
+    extra_context = {
+        'show_article': False,
+    }
+
+
+class LoreDetailView(DetailView):
+    model = Lore
+    template_name = 'internal/lore/lore_list.html'
+    context_object_name = 'shown_lore_article'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['show_article'] = True
+        context['all_lore_articles'] = Lore.objects.order_by('title')
+        return context
```

Comment:
I saw the previous comment, but frankly I don't understand the advantages of doing it the way you propose, so it's a bit frustrating to have to change something that works when I don't understand why I am changing it. I have now changed it, but I'd appreciate an explanation to why the second version of this code is better than the first.

Reply:
Sure, that's fair 🙂 So in this case, I guess the explanation has two parts (I'm guessing you were mostly interested in the first half of the second one..?):
1. "Why move context variables from `get_context_data()` to `extra_context`?"

   This is generally because there's less "noise" when reading it (compare e.g. `extra_context = {...}` to `def get_context_data(self, **kwargs): return {...}`), and so it's preferable whenever possible - which is the case when the values are constant (e.g. `True`), but not when they're variable/dynamic (e.g. `Lore.objects.order_by('title')`).

   In this case, however, one of the values is constant, while the other is variable, and so having _both_ `extra_context` and `get_context_data()` can make the code even noisier, but it can also be worth keeping the constant and variable variables separated, for the sake of organization - it's hard to say that either approach will always be the best choice 🤷 But since `LoreListView` already has an `extra_context` with `'show_article'`, I think it's best to have its "sister view", `LoreDetailView`, match that, if that makes sense :)
2. "Why structure the code inside `get_context_data()` the way I suggested?"

   This is purely to follow a principle of designing your code to be as future-proof as practically possible (i.e. as long as it doesn't take substantially more effort compared to a "naive" design). To illustrate why I think my suggestion adheres more closely to that principle, take the following example of code that could potentially be written in the future:
   * Writing a test for how the view and/or template behaves when some of the context variables are different (for whatever reason):
     ```python
     lore_detail_view_instance.get_context_data(show_article=False, all_lore_articles=None)
     ```
     This works with the code I suggested, but with the original code, `'show_article'` would have been overwritten (or "forced") to always have the value `True` regardless of the kwargs provided - and likewise for `'all_lore_articles'` being overwritten to equal `Lore.objects.order_by('title')`.

   That's what I meant when I said "[...] always let other code calling a context data method to be able to override the values set in the method, using `kwargs`" in the other comment.

   Also, as an added bonus, PyCharm seems to have a feature of providing code suggestions in templates for the context variables whose names are directly mentioned in the dict literal (i.e. `{...}`) returned by `get_context_data()` - but only when we write `return {...}` and not e.g. `context = {...}` followed by `return context`. (This is most likely because it's easier to parse one dict literal you know the location of (i.e. directly following the method's sole `return` statement), than to somehow analyze which context variables are inserted into `context` and then later returned 🤷)

Did that explanation make sense? If not, which details was I e.g. too vague about? 🙂